cves/CVE-2012-2774.yml

```diff
@@ -346,7 +365,12 @@ mistakes:
     Use those questions to inspire your answer. Don't feel obligated to answer
     every one. Write a thoughtful entry here that those ing the software
     engineering industry would find interesting.
-  answer:
+  answer: |
+    The biggest mistake seems to be neglecting an elementary part of threading.
+    Preventing a race condition is a check that is standard for most threaded
```

Comment:
Can you explain how the fix worked in a little more detail? How was the race condition prevented?